### ISSUE TITLE
Deprecation warnings since Typst 0.12 (use of measure and style)

### ISSUE THREAD

**Issue by brynhum** [jneug/typst-finite#7]
Finite 0.3.2 now generates warnings with Typst 0.12, related to the use of various functions/patterns involving measure and style that have become deprecated.

For example, the following code (from the documentation) generates a number of warnings:
```
#import "@preview/finite:0.3.2": automaton

#automaton((
  q0: (q1:0, q0:"0,1"),
  q1: (q0:(0,1), q2:"0"),
  q2: (),
))
```

Within finite itself:
```
warning: calling `measure` with a styles argument is deprecated
    ┌─ @preview/finite:0.3.2/util.typ:227:8
    │
227 │       m = measure(
    │ ╭─────────^
228 │ │       {
229 │ │         set text(s)
230 │ │         content
231 │ │       },
232 │ │       ctx.typst-style,
233 │ │     )
    │ ╰─────^
    │
    = hint: try removing the styles argument
```

And because of calls to outdated versions of cetz:

```
warning: `style` is deprecated
    ┌─ @preview/cetz:0.1.1/src/canvas.typ:259:47
    │
259 │               debug: false, body) = layout(ly => style(st => {
    │ ╭────────────────────────────────────────────────^
260 │ │   if body == none {
261 │ │     return []
262 │ │   }
    · │
353 │ │   )
354 │ │ }))
    │ ╰──^
    │
    = hint: use a `context` expression instead
```

**Comment by jneug** [comment]
I'm aware of the issue, but resolving it will need some work. Just updating CeTZ is not possible since the internals changed quite a bit and finite won't work with the new versions out of the box.

**Comment by brynhum** [comment]
Is there a guide somewhere about the changes in CeTZ and how to migrate code? Not that I know enough about Typst or CeTZ to make the changes but maybe someone else is able to contribute?

**Comment by jneug** [comment]
Just a heads up: I finally had the time to rewrite the package and a new version should be released soon.

**Comment by jneug** [comment]
Resolved in 0.4.0. Should be available in `@preview` soon.
